delta/plugins/search/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/SearchSpec.scala

```diff
@@ -0,0 +1,175 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.search
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri.Query
+import akka.testkit.TestKit
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.kernel.RetryStrategyConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.SparqlQuery.SparqlConstructQuery
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.CompositeViews
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeView
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewProjection.ElasticSearchProjection
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewSource.ProjectSource
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.ElasticSearchDocker.elasticsearchHost
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.ElasticSearchClient.Refresh
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.{ElasticSearchBulk, ElasticSearchClient}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.permissions
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.{ElasticSearchDocker, RemoteContextResolutionFixture}
+import ch.epfl.bluebrain.nexus.delta.plugins.search.Search.{ListProjections, TargetProjection}
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue.ContextObject
+import ch.epfl.bluebrain.nexus.delta.rdf.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.generators.{ProjectGen, ResourceGen}
+import ch.epfl.bluebrain.nexus.delta.sdk.http.{HttpClient, HttpClientConfig, HttpClientWorthRetry}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddress
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.{Group, User}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, Label, NonEmptySet}
+import ch.epfl.bluebrain.nexus.delta.sdk.testkit.{AclSetup, ConfigFixtures}
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.ExternalIndexingConfig
+import ch.epfl.bluebrain.nexus.testkit.{CirceLiteral, EitherValuable, IOValues, TestHelpers}
+import com.whisk.docker.scalatest.DockerTestKit
+import io.circe.{Json, JsonObject}
+import monix.bio.UIO
+import monix.execution.Scheduler
+import org.scalatest.concurrent.Eventually
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import org.scalatest.{CancelAfterFailure, Inspectors}
+
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.duration._
+
+class SearchSpec
```

Comment:
Search does not focus on the actual search but more on computing the accessible indices and we actually don't care about the query that we just pass on to the ES client.
So it would be great to be able to test only how we compute this list without having an ES client into this.
I think this is what we have done it somewhere for `AggregateElasticSearchViews`

Reply:
Filtering the views and making sure that the correct indices are used when using ES query are two main things here and they both need to be tested